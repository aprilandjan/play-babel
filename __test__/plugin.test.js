const babel = require('@babel/core');
const plugin = require('../plugin');

var example = `
function square(n) {
  return n * n;
};
`;

describe('plguin.js', () => {
  it('should transform correctly with no params', () => {
    const { code } = babel.transform(example, {
      plugins: [
        plugin
      ],
    });
    expect(code).toMatchSnapshot();
  });

  it('should transform correctly with customized params', () => {
    const { code } = babel.transform(example, {
      plugins: [
        [plugin, {
          replace: 'z',
        }]
      ],
    });
    expect(code).toMatchSnapshot();
  });
});
