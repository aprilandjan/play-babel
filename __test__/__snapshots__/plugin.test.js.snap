// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plguin.js should transform correctly with customized params 1`] = `
"function square(z) {
  return z * z;
}

;"
`;

exports[`plguin.js should transform correctly with no params 1`] = `
"function square(a) {
  return a * a;
}

;"
`;
